### ISSUE TITLE
Allow to save and use smalltalkCI result for deployment

### ISSUE THREAD

**Issue by zecke** [hpi-swa/smalltalkCI#148]
I use SmalltalkCI to build and test a Pharo image. My deployment is done using debian packages (i package the smalltalk image + scripts + runit config). For that I need to build an image.

It would be nice if run.sh takes an extra option to allow to save an image to a relative path and save it _after_ Metacello has loaded all packages and _before_ executing the tests.

```
$CI/run.sh -s Pharo-5.0 -o OsmoSmsc.image
$ ls -la OsmoSmsc.image | wc -l
1
```

Currently I copy the image out of the _build directory but that is saved _after_ the test has executed (e.g. might have some objects, garbage of the tests in it).

**Comment by fniephaus** [comment]
Interesting, maybe smalltalkCI should save an image after loading but before testing a project?
That would solve your problem and might be better anyway...

**Comment by zecke** [comment]
> On 29 May 2016, at 00:06, Fabio Niephaus <notifications@github.com> wrote:
> 
> Interesting, maybe smalltalkCI should save an image after loading but before testing a project?
> That would solve your problem and might be better anyway...

yes, it makes sense to save after loading has completed and before testing. It would be good to be able to specify the image name.

holger

**Comment by fniephaus** [comment]
I wonder: how do you "deploy" your images via debian packages? To where do you upload the bundle?
And have you seen #41?

**Comment by zecke** [comment]
I didn't see #41. I build debian source packages and upload them to either build.opensuse.org and a private appliance (depending on the kind of application). 

E.g. https://github.com/zecke/osmo-smsc/blob/master/deploy/deploy.sh here Henning uploads the source package to a nightly build.

**Comment by fniephaus** [comment]
Interesting, I'll have a look at your script. Do you only need an image and a changes file from smalltalkCI?

**Comment by zecke** [comment]
> On 29 May 2016, at 23:14, Fabio Niephaus <notifications@github.com> wrote:
> 
> Interesting, I'll have a look at your script. Do you only need an image and a changes file from smalltalkCI?

Exactly. I have (for Cog but not Spur yet) debian packages for the VM and sources file.

script: The auto-upload is a new thing , I use travis encrypt to encrypt the OBS_API/_PASS variable.
